payu/models/cice.py

```diff
@@ -162,29 +162,10 @@ def setup(self):
             self.ice_in.patch(history_nml)
 
         setup_nml = self.ice_in['setup_nml']
+        self._calc_runtime()
 
         if self.prior_restart_path:
-            # Generate ice.restart_file
-            # TODO: better check of restart filename
-            iced_restart_file = None
-            iced_restart_files = [f for f in self.get_prior_restart_files()
-                                  if f.startswith('iced.')]
-
-            if len(iced_restart_files) > 0:
-                iced_restart_file = sorted(iced_restart_files)[-1]
-
-            if iced_restart_file is None:
-                raise FileNotFoundError(
-                    f'No iced restart file found in {self.prior_restart_path}')
-
-            res_ptr_path = os.path.join(self.work_init_path,
-                                        'ice.restart_file')
-            if os.path.islink(res_ptr_path):
-                # If we've linked in a previous pointer it should be deleted
-                os.remove(res_ptr_path)
-            with open(res_ptr_path, 'w') as res_ptr:
-                res_dir = self.get_ptr_restart_dir()
-                print(os.path.join(res_dir, iced_restart_file), file=res_ptr)
+            self.make_restart_ptr()
```

Comment:
```suggestion
            self._make_restart_ptr()
```
I think these functions are intended to be private to this class ?

Reply:
Good point! I've added this in c1b3f57